Mohamed_Sahal/BinaryTree/Runner/BinaryTree.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.ArrayList;
+
+public class BinaryTree {
+
+    public static void main(String[] args) {
+        City seattle  =  new City("Seattle", 0);
+        City denver  =  new City("denver", 1305);
+        City miami  =  new City("miami", 2064);
+        City LosAngelos  =  new City("Los Angelos", 1016);
+        City dallas  =  new City("dallas", 1435);
+        City minneapols  =  new City("minneapolis", 990);
+        City newyork  =  new City("newyork", 2401);
+        City boston  =  new City("boston", 216);
+        //Node
+        Node root = new Node(seattle);
+
+        // Left Childen
+
+        root.left = new Node(denver);
+        root.left.left = new Node(miami);
+        root.left.right = new Node(LosAngelos);
+        root.left.right.left = new Node(dallas);
+        root.left.right.left.right = new Node(minneapols);
+
+        // Right
+
+        root.right = new Node(newyork);
+        root.right.right = new Node(boston);
+
+
+        BinaryTree.FindPath(root, "minneapolis");;
+        computeDistance();
+
+//        FindPathIterativly(root);;
+
+
+
+
+    }
+
+      /*
+        Recursive Solution
+     */
+
+    public static ArrayList<Object> RouteMiles = new ArrayList<>();
+
+    public static void computeDistance(){
+        int distance = 0;
+        for(int i = 0; i < RouteMiles.size(); i++){
+            int Mile = (int) RouteMiles.get(i);
+            distance += Mile;
+        }
+        System.out.println();
+        System.out.println("Distance - > "  + distance);
+    }
+
+    /*
+        Prints the Path cities to the destination.
+        FindPath : Stores Path using ArrayList Route
+        Parameter : root -> Starting Location , City - > Destination.
+     */
+
+
+    public static void FindPath(Node root, String city){
+        ArrayList<Object> Route = new ArrayList<>();
+        if(!CheckPath(root, Route, city)) {
+            System.out.println("No Path");
+        }
+        for (Object obj : Route){
```

Comment:
Here you could change it to `for (String str : Route)`.

Reply:
Thanks Mohamed. I will refactor.